internal/host/host.go

```diff
@@ -626,7 +633,10 @@ func (m *Manager) UpdateInstallProgress(ctx context.Context, h *models.Host, pro
 			previousProgress.CurrentStage, progress.CurrentStage, progress.ProgressInfo, extra...)
 	}
 	m.reportInstallationMetrics(ctx, h, previousProgress, progress.CurrentStage)
-	return err
+	if err != nil {
+		return common.NewApiError(http.StatusInternalServerError, err)
```

Comment:
I think just returning the error here is fine. I don't think we can assert this is definitely a 500 here.

Reply:
Gotcha, modified as such! Thanks!